hnn_core/network_models.py

```diff
@@ -0,0 +1,263 @@
+    lamtha = 70.
+    loc = 'soma'
+    for receptor in ['gabaa', 'gabab']:
+        key = f'gbar_L5Basket_{_short_name(target_cell)}_{receptor}'
+        weight = net._params[key]
+        net.add_connection(
+            src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    # layer2 Pyr -> layer5 Pyr
+    src_cell = 'L2_pyramidal'
+    lamtha = 3.
+    receptor = 'ampa'
+    for loc in ['proximal', 'distal']:
+        key = f'gbar_L2Pyr_{_short_name(target_cell)}'
+        weight = net._params[key]
+        net.add_connection(
+            src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    # layer2 Basket -> layer5 Pyr
+    src_cell = 'L2_basket'
+    lamtha = 50.
+    key = f'gbar_L2Basket_{_short_name(target_cell)}'
+    weight = net._params[key]
+    loc = 'distal'
+    receptor = 'gabaa'
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    # xx -> layer2 Basket
+    src_cell = 'L2_pyramidal'
+    target_cell = 'L2_basket'
+    lamtha = 3.
+    key = f'gbar_L2Pyr_{_short_name(target_cell)}'
+    weight = net._params[key]
+    loc = 'soma'
+    receptor = 'ampa'
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    src_cell = 'L2_basket'
+    lamtha = 20.
+    key = f'gbar_L2Basket_{_short_name(target_cell)}'
+    weight = net._params[key]
+    loc = 'soma'
+    receptor = 'gabaa'
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    # xx -> layer5 Basket
+    src_cell = 'L5_basket'
+    target_cell = 'L5_basket'
+    lamtha = 20.
+    loc = 'soma'
+    receptor = 'gabaa'
+    key = f'gbar_L5Basket_{_short_name(target_cell)}'
+    weight = net._params[key]
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha,
+        allow_autapses=False)
+
+    src_cell = 'L5_pyramidal'
+    lamtha = 3.
+    key = f'gbar_L5Pyr_{_short_name(target_cell)}'
+    weight = net._params[key]
+    loc = 'soma'
+    receptor = 'ampa'
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    src_cell = 'L2_pyramidal'
+    key = f'gbar_L2Pyr_{_short_name(target_cell)}'
+    weight = net._params[key]
+    loc = 'soma'
+    receptor = 'ampa'
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    return net
+
+
+def law_model(params=None, add_drives_from_params=False):
+    """Instantiate the beta modulated ERP network model.
+
+    Parameters
+    ----------
+    params : dict | None
+        The parameters to use for constructing the network.
+        If None, parameters loaded from beta_erp.json
+        Default: None
+    add_drives_from_params : bool
+        If True, add drives as defined in the params-dict. NB this is mainly
+        for backward-compatibility with HNN GUI, and will be deprecated in a
+        future release. Default: False
```

Comment:
this model is not in the HNN GUI, no? I think this comment holds for `Network` but elsewhere, the parameter will stay ...

Reply:
Yeah it's not provided in the GUI. Honestly it doesn't make much sense for this model since the "default drives" are demonstrated in the example. The drives defined in `default_params.json` are not tuned for this model and produce ERP's that are not reflective of real EEG/MEG.

Maybe just remove the parameter from `law_model`?

Reply:
While technically possible I think it somewhat goes against the goal of deprecating the params file. In HNN-GUI updating the params file based on user input is the only way to quickly reproduce model simulations. But we're not stuck with that problem in hnn-core.

Reply:
The way `network_models.py` is set up makes modifications to the base model really explicit, it's much more difficult to see how the models differ by comparing params files

Reply:
we're finally beginning to enter the regime of "object oriented programming" :) So just to be clear, the goal in `network_models` is to provide specific instantiations of the model? No reading/writing from files, correct? You take the default model and change it in a very specific way. If user wants to make further modifications, they do it through the API. I think then I'm in favor of removing the argument altogether.